azurerm/internal/services/network/data_source_network_service_tags.go

```diff
@@ -0,0 +1,105 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+)
+
+func dataSourceNetworkServiceTags() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNetworkServiceTagsRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"location": azure.SchemaLocation(),
+
+			"service": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"location_filter": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				StateFunc:        azure.NormalizeLocation,
+				DiffSuppressFunc: azure.SuppressLocationDiff,
+			},
+
+			"address_prefixes": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+		},
+	}
+}
+
+func dataSourceNetworkServiceTagsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Network.ServiceTagsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	res, err := client.List(ctx, location)
+	if err != nil {
+		return err
+	}
+
+	if res.Values == nil {
+		return errors.New("unexpected nil value for service tag information")
+	}
+
+	service := d.Get("service").(string)
+	locationFilter := d.Get("location_filter").(string)
+
+	for _, sti := range *res.Values {
+		if sti.Name == nil || !isServiceTagOf(*sti.Name, service) {
+			continue
+		}
+
+		if props := sti.Properties; props != nil {
+			if *props.Region == locationFilter {
+				addressPrefixes := make([]string, 0)
+				if props.AddressPrefixes != nil {
+					addressPrefixes = *props.AddressPrefixes
+				}
+				err = d.Set("address_prefixes", addressPrefixes)
+				if err != nil {
```

Comment:
I think this would be better as:
```suggestion
				if props.AddressPrefixes != nil {
					if err = d.Set("address_prefixes", *props.AddressPrefixes); err != nil {
						return fmt.Errorf("Error setting `address_prefixes`: %+v", err)
```

Reply:
@katbyte 
The reason to set `address_prefixes` anyway is to ensure there is always something in the `ResourceData` (even an empty list), so that (e.g.) output var could safely refer to `address_prefixes`. Otherwise, if you don't set it, terraform might panic if someone tries to refer to it.

Reply:
Would like to go here with @magodo proposal